R/input-submit.R

```diff
@@ -4,6 +4,54 @@
 #' button do not automatically update their outputs when inputs change,
 #' rather they wait until the user explicitly clicks the submit button.
 #'
+#' @section Warning:
+#' Submit buttons are very particular shiny objects. One the weirdest
+#' things about them is they have no IDs, and there is no way to regulate
+#' which inputs they should (and which ones they should not) be applied
+#' to. This means they are very unwieldy for anything but the simplest
+#' apps (see the example below). For example, having \emph{two} submit
+#' buttons in the same app will probably not do what you'd expect because
+#' submit buttons are not isolated in any way, so activating any of the
+#' two buttons will cause all inputs in the app to fire. Another problem
+#' with submit buttons is that dynamically created submit buttons (for
+#' example, with \code{\link{renderUI}} or \code{\link{insertUI}}) will
+#' not work.
+#'
+#' For all of these reasons, \strong{our general reccomendation is to
+#' avoid submit buttons} and
+#' \href{http://shiny.rstudio.com/articles/action-buttons.html#pattern-2---delay-reactions}{
+#' use action buttons instead when you want to delay a reaction}. Any
+#' code that uses a submit button can be converted to code that uses an
+#' action button instead (while the reverse is almost never true). So,
+#' if you find your app getting increasingly complicated, opt for using
+#' an action button (or more since there is no problem with multiple
+#' action buttons in the same app). The link above shows an example of
+#' a delayed reaction using an action button. For another example, the
+#' code below converts the example in this page (at the bottom) to an
+#' equivalent app using an action button (using the same pattern
+#' described in the link above):
+#'
+#' \preformatted{
+#' shinyApp(
```

Comment:
I think it's best to leave out this example, because I don't think people will understand how it's a translation of the original `submitButton` version -- it's notably more complex, and uses a function (`eventReactive`) they're likely not familiar with. I think the people will be better served by reading the article. For the same reason, I also changed the article link (above) to point to the top level of the actionButton article instead of jumping to the `eventReactive` example.